src/test/java/com/acrolinx/proxy/AcrolinxProxyFailureTest.java

```diff
@@ -11,8 +11,11 @@
 import jakarta.servlet.ServletException;
 import java.io.IOException;
 import java.net.ServerSocket;
+import java.util.concurrent.TimeUnit;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
 
+@Timeout(value = 5, unit = TimeUnit.SECONDS)
```

Comment:
What does this timeout do?

Reply:
See https://junit.org/junit5/docs/current/user-guide/#writing-tests-declarative-timeouts

We have a couple of "unhappy" tests which try to connect to a wrong domain/port/protocol. If we configure the timeout for the Apache HttpClient incorrectly, the test may run much longer or get stuck forever. This JUnit timeout makes sure that any test method inside this class will fail if it runs longer than 5 seconds, so it sets an upper boundary in error cases which is helpful.

Reply:
Update: I just renamed the method to `embeddedTomcatIntegrationTest`, thanks @willynganga for the suggestion.

Reply:
Makes sense, thank you!